### PR TITLE
SRCH-3674 Implement support for I14y and blended searches for redesigned SERP

### PR DESCRIPTION
**Pull request by ddzz**
## Summary
This PR modifies the SearchesController#index action to load the redesigned SERP for I14y and blended searches.
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review. If you cannot complete an item below, replace the checkbox with the ⚠️ `:warning:` emoji and explain why the step was not completed.
 
#### Functionality Checks
 
- [x] You have merged the latest changes from the target branch (usually `main`) into your branch.
  
- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket.

- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release - SRCH-####, SRCH-####, SRCH-####** matching the Jira ticket numbers in the release.
 
- [x] Automated checks pass. If Code Climate checks do not pass, explain reason for failures:
 
#### Process Checks

- [x] You have specified at least one "Reviewer".

### CODE REVIEW

File: app/controllers/searches_controller.rb [hunk 1]
```diff
@@ -63,23 +67,17 @@ def advanced
 
   private
 
-  # Temporarily disabling this cop until this method is refactored in
-  # upcoming changes for the SERP redesign.
-  # rubocop:disable Metrics/MethodLength
-  def pick_klass_vertical_template_layout
+  def pick_klass_vertical_template
```
Comment (ddzz) [inline app/controllers/searches_controller.rb:70]: This PR reverts temporary changes made to this method in the first SERP redesign PR.

File: app/controllers/searches_controller.rb [hunk 2]
```diff
@@ -1,6 +1,8 @@
 # frozen_string_literal: true
 
 class SearchesController < ApplicationController
+  layout :set_layout, only: [:index]
```
Comment (ddzz) [inline app/controllers/searches_controller.rb:4]: For now, the redesigned layout will be limited to the `#index` action. The other actions will be addressed in future PRs, and eventually the `only: [:index]` will be removed.

Comment (danswick) [reply app/controllers/searches_controller.rb:4]: Not a super strong opinion, but I tend to prefer to add comments for temporary code.

Comment (ddzz) [reply app/controllers/searches_controller.rb:4]: This is actually going to be removed very soon, possibly in the next redesign PR, so I think we're fine without a comment in this case.

**Pull request merged**